framework/src/test/java/com/epam/ta/page/CloudHomePage.java

```diff
@@ -0,0 +1,35 @@
+package com.epam.ta.page;
+
+import com.epam.ta.model.Search;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class CloudHomePage extends AbstractPage {
+
+    private static final String HOMEPAGE_URL = "https://cloud.google.com";
+
+    @FindBy(name = "q")
+    private WebElement searchBox;
+
+    public CloudHomePage(WebDriver driver) {
+        super(driver);
+    }
+
+    public CloudHomePage openPage() {
+        driver.get(HOMEPAGE_URL);
+        return this;
+    }
+
+    public CloudHomePage enterTermInTneSearchBox(Search search) {
+        waitForElementVisibilityOf(searchBox);
+        searchBox.sendKeys(search.getTerm());
+        return this;
+    }
+
+    public SearchResultPage searchForTerm() {
```

Comment:
Please adjust the method name. Something like 'submitSearchBoxRequest' would be better